### PR TITLE
Using streams

### PR DESCRIPTION
**Pull request by IfeSunmola**
Refactored some methods from using foreach loops to streams

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/owner/Owner.java [hunk 1]
```diff
@@ -130,17 +125,12 @@ public Pet getPet(Integer id) {
...
-				if (compName.equals(name)) {
-					return pet;
-				}
-			}
-		}
-		return null;
+		final String tempName = name.toLowerCase();
+		return getPets().stream()
+				.filter(pet -> !ignoreNew || !pet.isNew())
+				.filter(pet -> pet.getName() != null && pet.getName().toLowerCase().equals(tempName))
```
Comment (andrzejsydor) [inline src/main/java/org/springframework/samples/petclinic/owner/Owner.java:131]: you can use `equalsIgnoreCase`

File: src/main/java/org/springframework/samples/petclinic/owner/PetTypeFormatter.java [hunk 2]
```diff
@@ -50,13 +50,11 @@ public String print(PetType petType, Locale locale) {
...
 	@Override
 	public PetType parse(String text, Locale locale) throws ParseException {
-		Collection<PetType> findPetTypes = this.owners.findPetTypes();
-		for (PetType type : findPetTypes) {
-			if (type.getName().equals(text)) {
-				return type;
-			}
-		}
-		throw new ParseException("type not found: " + text, 0);
+		return owners.findPetTypes()
```
Comment (andrzejsydor) [inline src/main/java/org/springframework/samples/petclinic/owner/PetTypeFormatter.java:53]:
```suggestion
		return owners.findPetTypes().stream()
```

**Review by andrzejsydor — CHANGES_REQUESTED** [review]

**Comment by dsyer** [comment]
I'm not sure the stream API is any better for these tiny examples, and might even be worse - it just looks "clever" to me, not really self-documenting or anything useful. I'm tempted to just close this PR, but interested to hear other opinions.

**Comment by IfeSunmola** [comment]
> I'm not sure the stream API is any better for these tiny examples, and might even be worse - it just looks "clever" to me, not really self-documenting or anything useful. I'm tempted to just close this PR, but interested to hear other opinions.

Hi, 

I'm assuming this project is somewhat directed towards people that are new to java and spring? If that's the case, isn't it better for them to know that streams exists, and see the usage? Anyone that has made it here probably knows how to use the good ol for loop. 

Streams have been around for a relatively long time and questions about it even comes up in interviews. I didn't know about them till like a year ago because almost every code on the internet is still stuck on for loops

**Comment by dsyer** [comment]
And that’s kind of my point right there. The purpose of this project is definitely to educate, but the subject is Spring and Java, not the JDK. If for loops work and are understandable by a larger audience, that’s a good reason not to change anything. There really is no upside.

**Pull request closed without merging**